plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -78,6 +79,14 @@ class FeatureDevSessionContext(val project: Project, val maxProjectSizeBytes: Lo
         ignorePatternsWithGitIgnore = (ignorePatterns + parseGitIgnore().map { Regex(it) }).toList()
     }
 
+    fun isFileExtensionAllowed(file: File): Boolean {
+        // if it is a directory, it is allowed
+        if (file.isDirectory) return true
```

Comment:
`VirtualFile#isDirectory` exists

Reply:
Let me refactor this, there is no need to cast to `File` type